assets/js/scripts.js

```diff
@@ -25,4 +27,49 @@ window.onscroll = function() {
     document.getElementById("navbar").style.top = -1 * navBarHeight + "px";
   }
   prevScrollPos = currentScrollPos;
-}
\ No newline at end of file
+}
+
+
+//====animate portrait image to slide and fade in on scrolling====
+
+const image = document.querySelector('.profile-image');
+
+const imageObserver = new IntersectionObserver(entries => {
+  entries.forEach(entry => {
+    entry.target.classList.toggle("slide-in", entry.isIntersecting);
+    if(entry.isIntersecting) imageObserver.unobserve(entry.target);
+})},
+{
+  threshold: 0.5,
+});
+
+imageObserver.observe(image);
+
+
+//====animate cursor====
+
+const masked = document.querySelector('.masked');
+const cursorMovement = function (e) {
+  //gets mouse position
+  let x = e.pageX;
+  let y = e.pageY;
+
+  let cursorSize = '4em';
+
+  //centers masking cursor position on mouse position
+  masked.style.webkitMaskPositionX = 'calc(' + x + 'px - 0.5 * ' + cursorSize + ')';
+  masked.style.webkitMaskPositionY = 'calc(' + y + 'px - 0.5 * ' + cursorSize + ')';
+  masked.style.webkitMaskSize = cursorSize;
+
+  //makes cursor visible upon movement
+  masked.style.display = 'block';
+}
+
+//follow cursor on mousemove
+document.addEventListener('mousemove', cursorMovement);
+
+//cursor effects on mouseout
+document.addEventListener('mouseout', () =>{
+    cursorInner.style.display = 'none';
+    masked.style.webkitMaskSize = '0px';
+})

```

Comment:
It's best practice to have a new line at the end of each file for some reason

Reply:
Added